torch/_export/serde/serialize.py

```diff
@@ -1110,6 +1110,25 @@ def deserialize_sym_int(self, s: SymInt) -> Union[int, torch.SymInt]:
                             runtime_min=vr.lower,  # type: ignore[arg-type]
                             runtime_max=vr.upper  # type: ignore[arg-type]
                         )
+                else:
+                    # Placeholders, in particular, can have shapes as symbolic expressions.
+                    # We need to populate the shape env with the range constraints of their
+                    # free symbols, otherwise evaluating such expressions will error.
+                    self.symbol_name_to_symbol[val.expr_str] = sym
+                    free_symbols = sym.free_symbols
+                    for s in free_symbols:
+                        if s.name not in self.symbol_name_to_symbol:
+                            self.symbol_name_to_symbol[s.name] = s
+                        if vr := self.symbol_name_to_range.get(s.name):
+                            symbolic_shapes._constrain_symbol_range(
+                                self.shape_env,
+                                s,
+                                compiler_min=vr.lower,  # type: ignore[arg-type]
+                                compiler_max=vr.upper,  # type: ignore[arg-type]
+                                runtime_min=vr.lower,  # type: ignore[arg-type]
+                                runtime_max=vr.upper  # type: ignore[arg-type]
```

Comment:
This needs to be rebased past size-agnostic

Reply:
Oh, I was getting some new errors when I rebased, was wondering why.

Reply:
Just delete the runtime kwargs, should be enough